cli/pcluster/resources/batch/custom_resources_code/deregister_batch_mnp_job_definitions.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+#  the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+import re
+import time
+
+import boto3
+from botocore.exceptions import ClientError
+
+import crhelper
+
+# initialise logger
+logger = crhelper.log_config({"RequestId": "CONTAINER_INIT"}, loglevel="info")
+logger.info("Logging configured")
+# set global to track init failures
+init_failed = False
+
+try:
+    # Place initialization code here
+    logger.info("Container initialization completed")
+    batch_client = boto3.client("batch")
+except Exception as e:
+    logger.error(e, exc_info=True)
+    init_failed = e
+
+
+def get_job_definition_name_by_arn(job_definition_arn):
+    """
+    Parse Job Definition arn and get name.
+
+    Args:
+        job_definition_arn: something like arn:aws:batch:<region>:<account-id>:job-definition/<name>:<version>
+
+    Returns: the job definition name
+    """
+    pattern = r".*/(.*):(.*)"
+    return re.search(pattern, job_definition_arn).group(1)
+
+
+def retrieve_job_definition_revisions(name):
+    """
+    Retrieve all revisions for a given job definition.
+
+    Args:
+        name: name of the job definition
+
+    Returns: an array containing all job definition revisions ARNs
+    """
+    next_token = ""
+    job_definitions = []
+    while next_token is not None:
+        response = batch_client.describe_job_definitions(jobDefinitionName=name, nextToken=next_token, status="ACTIVE")
+        if "jobDefinitions" in response:
+            for job_definition in response["jobDefinitions"]:
+                job_definitions.append(job_definition["jobDefinitionArn"])
+        next_token = response.get("nextToken")
+        time.sleep(0.5)
```

Comment:
Could you add a comment to explain why we need the `sleep`?

Reply:
Done!